### PR TITLE
#7051: resblock ttnn submodule

### PR DESCRIPTION
**Pull request by saichandax**
This is a draft PR for the resBlock sub-module implementation of yolov4 and unit test.

### CODE REVIEW

**Comment by saichandax** [comment]
spdx license checks will be added to the files. Kindly ignore.

**Review by eyonland — APPROVED** [review]

**Review by dvartaniansTT — COMMENTED** [review]
Similar to the rest of the PR requests, please generate the torch graph and compare it to the original reference torch graph as well as the auto generated ttnn graph and attach them here.

example code to generate the torch graph:
`## Example usage
model = DownSample2()

input_tensor = torch.randn(1, 3, 320, 320) # Batch size of 1, 3 channels (RGB), 320x320 input - replace this line with the proper input size for your module
output_tensor = model(input_tensor)

model_graph = draw_graph(
model,
input_size=(1, 3, 320, 320),
dtypes=[torch.float32],
expand_nested=True,
graph_name="DownSample2",
depth=100,
directory=".",
)
model_graph.visual_graph.render(format="pdf")`

original reference graph:
[REPO_320x320_YOLOv4_PublicRepo.gv.pdf](https://github.com/tenstorrent-metal/tt-metal/files/14938896/REPO_320x320_YOLOv4_PublicRepo.gv.pdf)

**Review by dvartaniansTT — CHANGES_REQUESTED** [review]
Similar to the rest of the PR requests, please generate the torch graph and compare it to the original reference torch graph as well as the auto generated ttnn graph and attach them here.

example code to generate the torch graph:
`## Example usage
model = DownSample2()

input_tensor = torch.randn(1, 3, 320, 320) # Batch size of 1, 3 channels (RGB), 320x320 input - replace this line with the proper input size for your module
output_tensor = model(input_tensor)

model_graph = draw_graph(
model,
input_size=(1, 3, 320, 320),
dtypes=[torch.float32],
expand_nested=True,
graph_name="DownSample2",
depth=100,
directory=".",
)
model_graph.visual_graph.render(format="pdf")`

original reference graph:
[REPO_320x320_YOLOv4_PublicRepo.gv.pdf](https://github.com/tenstorrent-metal/tt-metal/files/14938896/REPO_320x320_YOLOv4_PublicRepo.gv.pdf)

**Comment by keerthana-r-mcw** [comment]
resblock
resblock_ttnn

**Comment by dvartaniansTT** [comment]
please generate the torch graph of your torch implementation and compare to the original graph. 
here is the code sample to do so [sample code](https://github.com/tenstorrent/tt-metal/blob/a34b544b7284d8126e9a5b47304fe1ad0ba56a39/tests/ttnn/integration_tests/yolov4/test_ttnn_downsample1.py#L190)

**Comment by dvartaniansTT** [comment]
> resblock resblock_ttnn

pls add the torch graph as well. torch graph of your implementation

File: tests/ttnn/integration_tests/yolov4/test_ttnn_resblock.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    # Convert output tensor back to Torch format
+    output_tensor = ttnn.to_torch(output_tensor)
+    output_tensor = output_tensor.reshape(1, 32, 32, 64)
+    output_tensor = torch.permute(output_tensor, (0, 3, 1, 2))
+    output_tensor = output_tensor.to(torch_input_tensor.dtype)
+
+    # Assertion
+    assert_with_pcc(torch_output_tensor, output_tensor, pcc=0.99)
+    ttnn.close_device(device)
```
Comment (dvartaniansTT) [inline tests/ttnn/integration_tests/yolov4/test_ttnn_resblock.py:125]: remove this line: `ttnn.close_device(device)`

File: tests/ttnn/integration_tests/yolov4/test_ttnn_resblock.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                    weight2, bias2, ttnn_module_args[f"resblock_{i}_conv2"], return_parallel_config=True
+                )
+
+        return parameters
+
+    return custom_preprocessor
+
+
+@skip_for_wormhole_b0()
+def test_resblock(reset_seeds):
```
Comment (dvartaniansTT) [inline tests/ttnn/integration_tests/yolov4/test_ttnn_resblock.py:64]: `def test_resblock(device, reset_seeds):`
and remove`device_id = 0`

**Review by dvartaniansTT — CHANGES_REQUESTED** [review]

**Comment by keerthana-r-mcw** [comment]
[resblock_pytoch.gv.pdf](https://github.com/tenstorrent/tt-metal/files/15007095/resblock_pytoch.gv.pdf)

**Review by dvartaniansTT — APPROVED** [review]

**Comment by saichandax** [comment]
Nightly fast dispatch: [passed](https://github.com/tenstorrent/tt-metal/actions/runs/8822603139)
All post commit: [passed](https://github.com/tenstorrent/tt-metal/actions/runs/8822593165/job/24223113440)

**Comment by saichandax** [comment]
post commit: [passed](https://github.com/tenstorrent/tt-metal/actions/runs/9007682768)
nightly fast dispatch: [failed](https://github.com/tenstorrent/tt-metal/actions/runs/9007680687)

**Comment by saichandax** [comment]
post commit: [passed](https://github.com/tenstorrent/tt-metal/actions/runs/9023890240)
nightly fast dispatch: [passed](https://github.com/tenstorrent/tt-metal/actions/runs/9023878276)

**Pull request merged**